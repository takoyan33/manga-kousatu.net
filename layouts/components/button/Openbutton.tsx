/* eslint-disable react/display-name */
import React from "react";
import { useEffect, useState, useMemo, useCallback } from "react";
import Avatar from "@mui/material/Avatar";

type Props = {
  text: string;
  onClick: () => void;
};

//React.memoåŒ–
const Openbutton: React.VFC<Props> = React.memo(({ text, onClick }) => {
  console.log(Openbutton);
  return (
    <div>
      <button
        onClick={onClick}
        type="button"
        className="bg-yellow-500 mt-2 p-1 inline-block text-white text-center"
      >
        {text}
      </button>
    </div>
  );
});

export default Openbutton;
