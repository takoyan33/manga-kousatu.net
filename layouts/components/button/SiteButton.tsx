import { DOMAttributes, FC, memo, ReactNode } from "react";
import Link from "next/link";
import Button from "@mui/material/Button";

type Props = {
  href: string;
  onClick?: DOMAttributes<HTMLButtonElement>["onClick"];
  // children: React.ReactNode;
  text: string;
  className: string;
  disabled?: boolean;
};

// eslint-disable-next-line react/display-name
export const SiteButton: React.VFC<Props> = memo(
  ({ href, onClick, text, className, disabled }) => {
    return (
      <div className={className}>
        <Link href={href}>
          <Button
            variant="outlined"
            onClick={onClick}
            disabled={disabled}
            type="submit"
          >
            {text}
          </Button>
        </Link>
      </div>
    );
  }
);

SiteButton.displayName = "SiteButton";
