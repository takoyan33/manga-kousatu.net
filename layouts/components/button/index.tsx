export * from "./loginbutton";
export * from "./Openbutton";
export * from "./registerbutton";
export * from "./SiteButton";
