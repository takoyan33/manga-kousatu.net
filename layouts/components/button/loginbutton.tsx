import React, { useContext } from "react";
// import { LoginContext } from "../../../pages/index";
import Link from "next/link";
import Button from "@mui/material/Button";

type Context = {
  text: string;
};

// eslint-disable-next-line react/display-name
export const Loginbutton = React.memo(() => {
  // const loginContext = useContext<Context>(LoginContext);
  return (
    <div>
      <Button variant="outlined" className="m-auto w-50 my-2">
        {/* <Link href="/login">{loginContext.text}</Link> */}
      </Button>
    </div>
  );
});
