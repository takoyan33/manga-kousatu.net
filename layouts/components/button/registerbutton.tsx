import React from "react";
import Link from "next/link";

export const Registerbutton = () => {
  return (
    <div>
      <button className="text-center m-auto my-2 border-solid  hover:border-dotted">
        <Link href="/register" className=" text-center m-auto my-2">
          新規登録はこちら
        </Link>
      </button>
    </div>
  );
};
