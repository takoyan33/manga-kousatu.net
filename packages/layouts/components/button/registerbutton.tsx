import React from "react";
import Link from "next/link";
import Button from "@mui/material/Button";

type Props = {
  text: string;
};

// eslint-disable-next-line react/display-name
export const Registerbutton: React.VFC<Props> = React.memo(({ text }) => {
  return (
    <div>
      <Button variant="outlined" className="m-auto w-50 my-2">
        <Link href="/register">{text}</Link>
      </Button>
    </div>
  );
});
