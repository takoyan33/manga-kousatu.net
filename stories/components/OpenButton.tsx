/* eslint-disable react/display-name */
import React from 'react'

interface buttonParams {
  text: string
  onClick?: () => void
}

//React.memoåŒ–
export const OpenButton = React.memo(({ text, onClick }: buttonParams) => {
  return (
    <button
      onClick={onClick}
      type='button'
      aria-label='open-button'
      className='open-button mt-2 inline-block bg-yellow-500 p-1 text-center text-white'
    >
      {text}
    </button>
  )
})
