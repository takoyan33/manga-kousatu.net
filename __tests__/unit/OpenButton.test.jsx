import { render, screen, fireEvent, userEvent } from '@testing-library/react'
import { OpenButton } from '../../layouts/components/button/OpenButton'
import '@testing-library/jest-dom'

describe('A1_Component > OpenButton', () => {
  it('[A1_4_1] レンダリングされているか', () => {
    render(<OpenButton text='test' />)
  })
  it('[A1_4_2] buttonをクリックできるか', () => {
    render(<OpenButton text='test' />)
    fireEvent.click(screen.getByRole('button', { name: 'open-button' }))
  })

  //[A1_3_4] LINKをクリックできるか
  // const passwordChangeLink = screen.getByRole('link', { name: 'password-change' });
  // fireEvent.click(passwordChangeLink);
})
