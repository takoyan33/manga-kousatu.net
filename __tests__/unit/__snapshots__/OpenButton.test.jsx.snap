// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A1_Component > OpenButton [A1_4_3] Pageのスナップショット 1`] = `
<div>
  <button
    aria-label="open-button"
    class="mt-2 inline-block bg-yellow-500 p-1 text-center text-white"
    type="button"
  >
    test
  </button>
</div>
`;
